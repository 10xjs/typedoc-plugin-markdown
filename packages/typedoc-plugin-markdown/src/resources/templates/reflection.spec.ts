import * as path from 'path';

import * as fs from 'fs-extra';
import * as Handlebars from 'handlebars';

import {
  TestApp,
  compileHandlabrs,
  stubHelpers,
  stubPartials,
} from '../../../test/utils';

describe(`[template] reflection.hbs`, () => {
  let testApp: TestApp;
  let template: Handlebars.TemplateDelegate;

  beforeAll(() => {
    testApp = new TestApp(['reflections']);
    testApp.bootstrap(false);
    stubPartials();
    stubHelpers();
    const partial = fs.readFileSync(path.resolve(__dirname, 'reflection.hbs'));
    template = Handlebars.compile(partial.toString());
  });

  beforeEach(() => {
    Handlebars.registerHelper('ifShowIndexes', (options) => options.fn());
  });

  test(`should compile reflection with type params / implemented by`, () => {
    expect(
      compileHandlabrs(
        template,
        testApp.project.findReflectionByName('TypeParamsInterface'),
      ),
    ).toMatchSnapshot();
  });

  test(`should compile a callable reflection`, () => {
    expect(
      compileHandlabrs(
        template,
        testApp.project.findReflectionByName('CallableInterface'),
      ),
    ).toMatchSnapshot();
  });

  test(`should compile an indexable reflection`, () => {
    expect(
      compileHandlabrs(
        template,
        testApp.project.findReflectionByName('IndexableInterface'),
      ),
    ).toMatchSnapshot();
  });

  test(`should compile reflection without index`, () => {
    Handlebars.registerHelper('ifShowIndexes', (options) => options.inverse());
    expect(
      compileHandlabrs(
        template,
        testApp.project.findReflectionByName('TypeParamsInterface'),
      ),
    ).toMatchSnapshot();
  });
});
