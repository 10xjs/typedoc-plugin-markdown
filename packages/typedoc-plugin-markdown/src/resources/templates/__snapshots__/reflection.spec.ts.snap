// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`[template] reflection.hbs should compile a callable reflection 1`] = `
"## Hierarchy

[partial: hierarchy]

## Callable

[partial: member.signature]

[partial: index]

[partial: members]
"
`;

exports[`[template] reflection.hbs should compile an indexable reflection 1`] = `
"## Hierarchy

[partial: hierarchy]

## Indexable

[partial: member.indexSignatures]

[partial: index]

[partial: members]
"
`;

exports[`[template] reflection.hbs should compile reflection with type params / implemented by 1`] = `
"[partial: comment]

## Type parameters

[partial: typeParameters]

## Hierarchy

[partial: hierarchy]

## Implemented by

* [helper: type]
* [helper: type]
* [helper: type]

[partial: index]

[partial: members]
"
`;

exports[`[template] reflection.hbs should compile reflection without index 1`] = `
"[partial: comment]

## Type parameters

[partial: typeParameters]

## Hierarchy

[partial: hierarchy]

## Implemented by

* [helper: type]
* [helper: type]
* [helper: type]

[partial: members]
"
`;
